packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -156,6 +157,37 @@ export const assertRequesterIsConsumerOrProducer = (
   }
 };
 
+export const assertRequesterIsConsumerOrProducerOrDelegate = async (
+  agreement: Agreement,
+  authData: AuthData,
+  readModelService: ReadModelService
+): Promise<void> => {
+  try {
+    assertRequesterIsConsumer(agreement, authData);
+  } catch (error) {
+    try {
+      assertRequesterIsProducer(agreement, authData);
+    } catch (error) {
+      const delegation = await readModelService.getDelegationByDelegateId(
+        authData.organizationId
+      );
```

Comment:
it could be moved to `assertRequesterIsDelegate`

Reply:
I need it outside for the https://github.com/pagopa/interop-be-monorepo/pull/1162